app/components/Poll/index.js

```diff
@@ -9,247 +9,225 @@ import { Link } from 'react-router-dom';
 import Icon from 'app/components/Icon';
 import { Flex } from 'app/components/Layout';
 import Tooltip from 'app/components/Tooltip';
-import cx from 'classnames';
 
 type Props = {
   poll: PollEntity,
   handleVote: (pollId: number, optionId: number) => Promise<*>,
   allowedToViewHiddenResults?: boolean,
   backgroundLight?: boolean,
-  truncate?: number,
   details?: boolean,
+  expanded?: boolean,
+  alwaysOpen?: boolean,
 };
 
 type OptionEntityRatio = OptionEntity & {
   ratio: number,
 };
 
-type State = {
-  truncateOptions: boolean,
-  shuffledOptions: Array<OptionEntityRatio>,
-  expanded: boolean,
+// As described in: https://stackoverflow.com/questions/13483430/how-to-make-rounded-percentages-add-up-to-100
+export const perfectRatios = (
+  options: $ReadOnlyArray<OptionEntityRatio>
+): OptionEntityRatio[] => {
+  const off =
+    100 - options.reduce((a, option) => a + Math.floor(option.ratio), 0);
+  return sortBy<OptionEntityRatio>(
+    options,
+    (o: OptionEntityRatio) => Math.floor(o.ratio) - o.ratio
+  )
+    .map((option: OptionEntityRatio, index: number) => {
+      return {
+        ...option,
+        ratio: Math.floor(option.ratio) + (index < off ? 1 : 0),
+      };
+    })
+    .sort((a, b) => b.ratio - a.ratio);
 };
 
-class Poll extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    const options = this.optionsWithPerfectRatios(props.poll.options);
-    const shuffledOptions = this.shuffle(options);
-    if (props.truncate && options.length > props.truncate) {
-      this.state = {
-        truncateOptions: true,
-        shuffledOptions: shuffledOptions,
-        expanded: false,
-      };
-    } else {
-      this.state = {
-        truncateOptions: false,
-        shuffledOptions: shuffledOptions,
-        expanded: true,
-      };
-    }
+const optionsWithPerfectRatios = (options: Array<OptionEntity>) => {
+  const totalVotes = options.reduce((a, option) => a + option.votes, 0);
+  const ratios = options.map((option) => {
+    return { ...option, ratio: (option.votes / totalVotes) * 100 };
+  });
+  return perfectRatios(ratios);
+};
+
+const shuffle = (array: Array<OptionEntityRatio>) => {
+  const oldArray = array.slice(0);
+  const newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    const randIndex = Math.floor(Math.random() * oldArray.length);
+    newArray[i] = oldArray[randIndex];
+    oldArray.splice(randIndex, 1);
   }
 
-  toggleTruncate = () => {
-    this.setState({
-      expanded: !this.state.expanded,
-    });
-  };
+  return newArray;
+};
 
-  optionsWithPerfectRatios = (options: Array<OptionEntity>) => {
-    const totalVotes = options.reduce((a, option) => a + option.votes, 0);
-    const ratios = options.map((option) => {
-      return { ...option, ratio: (option.votes / totalVotes) * 100 };
-    });
-    return this.perfectRatios(ratios);
-  };
+const Poll = ({
+  poll,
+  handleVote,
+  backgroundLight,
```

Comment:
`backgroundLight` seems to be unused. Remove?

Reply:
yeet